### PR TITLE
Clean orphaned cached accessories / handle uncaught exceptions

### PR DESCRIPTION
**Pull request by ebaauw**
See #2120:
- Add `-R` or `--remove-orphans` command line option;
- Clean orphaned cached accessores, for which no plugin was loaded, when `-R` was specified on the command line.

Handle uncaught exceptions to do a clean shutdown of homebridge instead of crashing NodeJS.

@nfarina, @KhaosT, please review.  Some technical notes:
- This change will unconditionally remove cached "stuff" that is not a `PlatformAccessory`.  I could add the check to do this only when `-R` was specified, but I see no reason why you would want to keep this "stuff" in the cache;
- I daren't remove elements from the `_cachedPlatformAccessories` while looping over them, hence the build up of `verifiedAccessories` to replace `_cachedPlatformAccessories` after the loop.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -42,5 +49,12 @@ module.exports = function() {
     });
   });
 
+  process.on('uncaughtException', function(error) {
```
Comment (KhaosT) [inline lib/cli.js]: Is this related to the clean cached accessories?

Comment (ebaauw) [reply lib/cli.js]: Not really, sorry.  When I commit a new change, GitHub adds this to the open PR.

It’s actually more related to the check for the number of bridged accessories in HAP-NodeJS, since that error is unchecked by homebridge.  I’ve had this logic in my homebridge-zp plugin for months, to make sure the subscriptions with the Sonos speakers are canceled before homebridge exits.  Only then, the error is logged by the plugin logger, making it seem like my plugin caused the exception instead of only handling it.  See https://github.com/ebaauw/homebridge-zp/issues/41 and https://github.com/ebaauw/homebridge-zp/issues/49.

**Comment by KhaosT** [comment]
LGTM Thanks! 🙏

**Comment by lboue** [comment]
Thanks a lot, worked fine for me with homebridge 0.4.48. I noticed that in log file:
```
Failed to find plugin to handle accessory Det. Fumee etage
Removing orphaned accessory Det. Fumee etage
```

+1
